pkgs/top-level/stage.nix

```diff
@@ -154,6 +154,19 @@ let
         };
       };
     };
+
+    # Extend the package set with zero or more overlays. This preserves
+    # preexisting overlays. Prefer to initialize with the right overlays
+    # in one go when calling Nixpkgs, for performance and simplicity.
+    appendOverlays = extraOverlays:
+      import ./stage.nix (args // { overlays = args.overlays ++ extraOverlays; });
```

Comment:
Move this function out-side of `otherPackagesSets` or make it an inner function for the `extend` function, such that it is not visible to users of Nixpkgs.

Reply:
Doesn't that hurt performance, in case someone needs to add more than one overlay at a time?

Reply:
Oh, I see what you mean. I thought of it only as an implementation short-cut and not as a function to be used directly. I guess you are right and this should remain exposed.